experimental/dispatch_profiler/matmul.py

```diff
@@ -0,0 +1,590 @@
+    vmfb_file = self.vmfb_benchmark_file if compilation_mode == CompilationMode.Benchmark else self.vmfb_verify_file
+
+    values = {
+        'iree_compile_path': self.iree_compile_path,
+        'source_mlir_file': self.source_mlir_file,
+        'device': self.device,
+        'benchmark-dispatch-repeat-count': str(benchmark_dispatch_repeat_count),
+        'vmfb_file': vmfb_file
+    }
+
+    if not os.path.exists(vmfb_file) or self.force_compile:
+      cmd = SubstituteTemplate(cmd_template, values)
+      print(">> Compilation command for " +
+            CompilationModeNames[compilation_mode] + " : " + cmd)
+      subprocess.getoutput(cmd)
+
+    else:
+      if self.verbose:
+        print("Skipping compilation of matmul operation: " + vmfb_file +
+              " since it already exists.")
+
+  def verify(self, configuration):
+    """Verifies the matmul operation with a given configuration."""
+    # First compile the operation to a vmfb file.
+    self.compile(CompilationMode.Verify)
+
+    # Template for the verification command using iree-run-module as the verification tool.
+    cmd_templete = \
+    "${iree_run_module_path} --module=${vmfb_file} --device=${device} --function=${function_name} "\
+    "--input=@${lhs_npy_file} --input=@${rhs_npy_file} "\
+    "--expected_output=@${expected_result_npy_file}"
+
+    # Function name operation + configuration.
+    function_name = self.operation.name() + "_" + configuration.name()
+
+    # Verify using random data distribution.
+    # TODO 1) make input distribution configurable through command line.
+    # TODO 2) make the reference run to check if reference npy files are present,
+    #         then do not re-run the reference.
+    reference_op = ReferenceMatmulOperation(self.operation,\
+                                         Distribution.Random,\
+                                         Distribution.Random)
+
+    lhs_npy_file = os.path.join(self.operation_path, reference_op.filename_lhs)
+    rhs_npy_file = os.path.join(self.operation_path, reference_op.filename_rhs)
+    expected_result_npy_file = os.path.join(
+        self.operation_path, reference_op.reference_filename_result)
+
+    # Run the reference implementation and generate npy files
+    reference_op.run_and_save(self.operation_path)
+
+    values = {
+        'iree_run_module_path': self.iree_run_module_path,
+        'vmfb_file': self.vmfb_verify_file,
+        'device': self.device,
+        'function_name': function_name,
+        'lhs_npy_file': lhs_npy_file,
+        'rhs_npy_file': rhs_npy_file,
+        'expected_result_npy_file': expected_result_npy_file
+    }
+    # Create the verification command.
+    cmd = SubstituteTemplate(cmd_templete, values)
+    if self.verbose:
+      print(">> Verification command: " + cmd)
+
+    # Launch verification.
+    cmd_output = subprocess.getoutput(cmd)
```

Comment:
Instead of passing a string with all command arguments, it's usually recommended to use https://docs.python.org/3/library/subprocess.html#subprocess.check_output and pass a list of arguments, like:
```
subprocess.check_output([
  self.iree_run_module_path,
  f'--module={vmfb_file}',
  f'--device={self.device}',
  ...
])
```

That avoids the problems of quoting characters in the command lines. For example, if the path `lhs_npy_file` contains whitespaces, like "C:\Program (x86)", the `--input=` in `cmd` will become `--input=C:\Program (x86)`, where the `(x86)` incorrectly becomes a new positional argument.

We can fix this later.

Reply:
Done. using `suprocess.check_output.